docs/source/en/core/development.md

```diff
@@ -1 +1,339 @@
+$ npm test -- --help
+$
+$ # Equivalent to `TESTS=test/**/test.js npm test`. Since the limitation of bash, it's better to add double qoute to path.
+$ npm test "test/**/test.js"
+$
+$ # Equivalent to `TEST_REPORTER=dot npm test`
+$ npm test -- --reporter=dot
+$
+$ # Parameters of mocha are supported, such as grep, require, etc.
+$ npm test -- -t 30000 --grep="should GET"
+```
+
+## Code Coverage
+
+egg-bin has build-in [nyc](https://github.com/istanbuljs/nyc) to support calculating code coverage report of unit testing.
+
+Add `npm scripts` into `package.json`：
+
+```json
+{
+  "scripts": {
+    "cov": "egg-bin cov"
+  }
+}
+```
+
+And then we can get code coverage report of unit testing via command `npm run cov`.
+
+```bash
+$ egg-bin cov
+
+  test/controller/home.test.js
+    GET /
+      ✓ should status 200 and get the body
+    POST /post
+      ✓ should status 200 and get the request body
+
+  ...
+
+  16 passing (1s)
+
+=============================== Coverage summary ===============================
+Statements   : 100% ( 41/41 )
+Branches     : 87.5% ( 7/8 )
+Functions    : 100% ( 10/10 )
+Lines        : 100% ( 41/41 )
+================================================================================
+```
+
+And we may open HTML file of complete code coverage report via command `open coverage/lcov-report/index.html`.
+
+![image](https://cloud.githubusercontent.com/assets/156269/21845201/a9a85ab6-d82c-11e6-8c24-5e85f352be4a.png)
+
+### Config Environment
+
+Same as `test` script, to run `cov`, environment need to be set to `env: unittest` as well. The configuration is also an object combined from `config.local.js` and `config.unittest.js` files.
+
+### Ignore Files
+
+To ignore some files in code coverage rate calculation, you may use `COV_EXCLUDES` environment variable to achieve this:
+
+```bash
+$ COV_EXCLUDES=app/plugins/c* npm run cov
+$ # Or pass this setting via parameters
+$ npm run cov -- --x=app/plugins/c*
+```
+
+## Debug
+
+### Print Logs
+
+### Use Logger Module
+
+There's built-in [Log](./logger.md) in egg. You may use `logger.debug()` to print out debug information. **It is recommended to be used in your implementation.**
+
+```js
+// controller
+this.logger.debug('current user: %j', this.user);
+
+// service
+this.ctx.logger.debug('debug info from service');
+
+// app/init.js
+app.logger.debug('app init');
+```
+
+Levels of logs can be configured via `config.logger.level` for printing into file, and `config.logger.consoleLevel` for printing into console.
+
+### Use Debug Module
+
+[debug](https://www.npmjs.com/package/debug) module is a debug tool which is widely adopted in Node.js community. Plenty of modules are using it to print debug information. Egg community has widely adopted it as well. **It is recommended to be used in framework and plugin development**
+
+To watch on the running process, we may start with certain code for debugging via `DEBUG` enviroment variable.
```

Comment:
It's easy for us to watch the whole process of test through 'DEBUG` as the environment variable to start with certain code.